src/server/main_service.cc

```diff
@@ -939,6 +939,8 @@ void Service::DispatchCommand(CmdArgList args, facade::ConnectionContext* cntx)
               << " in dbid=" << dfly_cntx->conn_state.db_index;
   }
 
+  etl.AwaitPauseState(/*is_write=*/cid->opt_mask() & CO::WRITE);
+
```

Comment:
This will produce inconsistent results with multi. MULTI/EXEC and many script will run till the end, yet some scripts that use DispatchCommand will block here 

I guess the intention is to let all multi run till the end - just check !under_script

Reply:
And it does not block DispatchManyCommands (which is probably not intended)

Reply:
from redis doc: that interaction with replicas should continue normally 
can we check if the connection is from a replica and if so allow the command execution?